diff -up mutt-1.5.17/smime_keys.pl.smimekeys mutt-1.5.17/smime_keys.pl
--- mutt-1.5.17/smime_keys.pl.smimekeys	2007-03-02 19:44:44.000000000 +0100
+++ mutt-1.5.17/smime_keys.pl	2008-04-04 12:33:34.000000000 +0200
@@ -20,6 +20,7 @@
 
 use strict;
 use File::Copy;
+use File::Glob ':glob';
 
 umask 077;
 
@@ -225,7 +226,7 @@ You must set this in your mutt config fi
 EOF
 #'
 
-    $answer =~ /\"(.*?)\"/ and return $1;
+    $answer =~ /\"(.*?)\"/ and return bsd_glob($1, GLOB_TILDE | GLOB_NOCHECK);
     
     $answer =~ /^Mutt (.*?) / and die<<EOF;
 This script requires mutt 1.5.0 or later. You are using mutt $1.
