diff --git a/mutt_ssl_gnutls.c b/mutt_ssl_gnutls.c
index 970a9c9..6d4e2f6 100644
--- a/mutt_ssl_gnutls.c
+++ b/mutt_ssl_gnutls.c
@@ -445,7 +445,8 @@ static int tls_compare_certificates (const gnutls_datum *peercert)
     }
 
     /* find start of cert, skipping junk */
-    ptr = (unsigned char *)strstr((char*)b64_data.data, CERT_SEP);
+    ptr = (unsigned char *)memmem((char*)b64_data.data, b64_data.size,
+                                  CERT_SEP, sizeof(CERT_SEP));
     if (!ptr)
     {
       gnutls_free(cert.data);
@@ -453,8 +454,10 @@ static int tls_compare_certificates (const gnutls_datum *peercert)
       return 0;
     }
     /* find start of next cert */
-    ptr = (unsigned char *)strstr((char*)ptr + 1, CERT_SEP);
-
+    ptr += 1;
+    ptr = (unsigned char *)memmem((char*)ptr,
+                                  b64_data.size - (ptr - b64_data.data),
+                                  CERT_SEP, sizeof(CERT_SEP));
     b64_data.size = b64_data.size - (ptr - b64_data.data);
     b64_data.data = ptr;
 
