diff -up mutt-1.5.17/doc/Makefile.in.manual mutt-1.5.17/doc/Makefile.in
--- mutt-1.5.17/doc/Makefile.in.manual	2007-11-01 20:13:14.000000000 +0100
+++ mutt-1.5.17/doc/Makefile.in	2007-11-02 13:00:53.000000000 +0100
@@ -481,10 +481,9 @@ muttrc.man: ../makedoc$(EXEEXT) $(top_sr
 mutt.1: $(srcdir)/mutt.man
 	$(EDIT) $(srcdir)/mutt.man > $@
 
-stamp-doc-xml: $(top_srcdir)/makedoc.c $(top_srcdir)/makedoc-defs.h $(top_srcdir)/init.h \
+stamp-doc-xml: ../makedoc$(EXEEXT) $(top_srcdir)/makedoc-defs.h $(top_srcdir)/init.h \
                manual.xml.head $(top_srcdir)/functions.h $(top_srcdir)/OPS* manual.xml.tail \
                $(srcdir)/gen-map-doc $(top_srcdir)/VERSION $(top_srcdir)/ChangeLog
-	$(MAKE) ../makedoc$(EXEEXT) # we do not want to rebuild the documentation in tarball builds
 	( date=`head -n 1 $(top_srcdir)/ChangeLog | LC_ALL=C cut -d ' ' -f 1` && \
 	  sed -e "s/@VERSION\@/`cat $(top_srcdir)/VERSION` ($$date)/" $(srcdir)/manual.xml.head && \
 	  $(MAKEDOC_CPP) $(top_srcdir)/init.h | ../makedoc -s && \
diff -up mutt-1.5.17/init.h.manual mutt-1.5.17/init.h
--- mutt-1.5.17/init.h.manual	2007-10-31 18:48:52.000000000 +0100
+++ mutt-1.5.17/init.h	2007-11-02 13:00:53.000000000 +0100
@@ -19,7 +19,6 @@
 
 #ifdef _MAKEDOC
 # include "config.h"
-# include "makedoc-defs.h"
 #else
 # include "sort.h"
 #endif
