--- urlview-0.9/url_handler.sh.default	Tue Jul  4 06:14:30 2000
+++ urlview-0.9/url_handler.sh	Tue Jul  4 13:02:40 2000
@@ -28,11 +28,11 @@
 # VT: Launch in the same terminal
 
 # The lists of programs to be executed are
-https_prgs="/usr/X11R6/bin/netscape:XW /usr/bin/lynx:XT"
-http_prgs="/usr/bin/lynx:XT /usr/X11R6/bin/netscape:XW"
+https_prgs="/usr/bin/htmlview:XW /usr/bin/mozilla:XW /usr/bin/links:XT /usr/bin/lynx:XT"
+http_prgs="/usr/bin/htmlview:XW /usr/bin/mozilla:XW /usr/bin/links:XT /usr/bin/lynx:XT"
 mailto_prgs="/usr/bin/mutt:VT /usr/bin/elm:VT /usr/bin/pine:VT /usr/bin/mail:VT"
-gopher_prgs="/usr/bin/lynx:XT /usr/bin/gopher:XT"
-ftp_prgs="/usr/bin/lynx:XT /usr/bin/ncftp:XT"
+gopher_prgs="/usr/bin/lynx:XT"
+ftp_prgs="/usr/bin/ncftp:XT /usr/bin/wget:XT /usr/bin/lynx:XT"
 
 # Program used as an xterm (if it doesn't support -T you'll need to change
 # the command line in getprg)
